composer/utils/fx_utils.py

```diff
@@ -51,7 +54,15 @@ def count_op_instances(gm: GraphModule, ops: Union[Callable, str, List[Union[Cal
         int: The number of instances of ``ops`` in ``gm``
     """
     ops = list(ensure_tuple(ops))
-    return sum(any(n.target == op for op in ops) for n in gm.graph.nodes)
+    all_modules = dict(gm.named_modules())
+    count = 0
+    for n in gm.graph.nodes:
+        for op in ops:
+            if n.target == op:
+                count += 1
+            elif n.op == 'call_module' and isinstance(all_modules[n.target], op):
```

Comment:
I'm a bit confused here when this isinstance check would return True if `op` is a string or a generic function. For example, the following errors:

```python
>>> isinstance(2, "hi")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union
```

Reply:
Thanks for catching this error. This check should have only been executed when `op` is a class. Fixed it. It counts the instances of nn.Modules. For example, 

```
>>> class M(torch.nn.Module):
...   def __init__(self):
...     self.fc = nn.Linear(8, 8)
...     super().__init__()
...   def forward(self, x):
...     return self.fc(x)
>>> module = M()
>>> traced = symbolic_trace(module)
>>> count_op_instances(traced, torch.add)
0
>>> count_op_instances(traced, nn.Linear)
1
>>> count_op_instances(traced, "hi")
0
>>> count_op_instances(traced, nn.ReLU)
0
```